pkg/node/manager.go

```diff
@@ -191,6 +192,12 @@ func replaceNodeRoute(ip *net.IPNet) {
 	}
 
 	route := netlink.Route{LinkIndex: link.Attrs().Index, Dst: ip, Gw: via, Src: local}
+	// If the route includes the local address, then the route is for
+	// local containers and we can use a high MTU for transmit. Otherwise,
+	// it needs to be able to fit within the MTU of tunnel devices.
+	if !ip.Contains(local) {
```

Comment:
Nice, this is beautiful

Reply:
I retained this, as routes to non-local destinations should actually use the lower MTU. It doesn't affect endpoint MTU, however.